components/fxa-client/src/oauth.rs

```diff
@@ -122,6 +122,40 @@ impl FirefoxAccount {
         self.oauth_flow(url, &scopes, wants_keys)
     }
 
+    /// Initiate a `force_auth` OAuth login flow providing an email value return a URL that should be navigated to.
+    ///
+    /// * `email` - The email that is used for `force_auth`
+    /// * `scopes` - Space-separated list of requested scopes.
+    /// * `wants_keys` - Retrieve scoped keys associated with scopes supporting it.
+    pub fn force_auth_oauth_flow(
+        &mut self,
+        email: &str,
+        scopes: &[&str],
+        wants_keys: bool,
+    ) -> Result<String> {
+        let mut url = self.state.config.authorization_endpoint()?;
+        url.query_pairs_mut()
+            .append_pair("action", "force_auth")
+            .append_pair("email", email)
+            .append_pair("response_type", "code");
+
+        let scopes: Vec<String> = match self.state.refresh_token {
+            Some(ref refresh_token) => {
+                // Union of the already held scopes and the one requested.
+                let mut all_scopes: Vec<String> = vec![];
+                all_scopes.extend(scopes.iter().map(ToString::to_string));
+                let existing_scopes = refresh_token.scopes.clone();
+                all_scopes.extend(existing_scopes);
+                HashSet::<String>::from_iter(all_scopes)
+                    .into_iter()
+                    .collect()
+            }
+            None => scopes.iter().map(ToString::to_string).collect(),
+        };
+        let scopes: Vec<&str> = scopes.iter().map(<_>::as_ref).collect();
+        self.oauth_flow(url, &scopes, wants_keys)
+    }
```

Comment:
One improvement we could make here, would be to explicitly tag the flow as a "force-auth" flow in self.flow_store. Then `complete_oauth_flow` can double-check whether the user did indeed re-auth as the intended user account, which might help us detect or avoid some unexpected edge-cases.